src/yb/yql/cql/ql/util/cql_message.h

```diff
@@ -131,6 +133,32 @@ class CQLMessage {
        : version(version), flags(0), stream_id(stream_id), opcode(opcode) { }
   };
 
+  // The CQL metadata consists of the first 8 bytes of the request body.
+  // The format of the request metadata is
+  //   0        16        32        48        64
+  //   +---------+---------+---------+---------+
+  //   |           unused            | Q limit |
+  //   +---------+---------+---------+---------+
+  // The format of the response metadata is
+  //   0        16        32        48        64
+  //   +---------+---------+---------+---------+
+  //   |           unused            |  Q pos  |
+  //   +---------+---------+---------+---------+
+  //
+  // The RPC queue limit ("Q limit") is a unsigned 16-bit integer indicating
+  // whether the request should be dropped. If the queue position of the
+  // incoming RPC is greater than the queue limit, then the request gets
+  // dropped immediately and an ERROR_SERVER_TOO_BUSY response is sent back
+  // to the client. If the queue limit is 0, then the request is not dropped
+  // based on the queue position.
+  //
+  // The RPC queue position ("Q pos") is a signed 16-bit integer indicating
+  // the queue position of the corresponding inbound RPC. If for some reason
+  // the queue position could not be determined, then a value of -1 is returned.
+  static constexpr size_t kMetadataSize = 8;
+  static constexpr size_t kMetadataQueueLimitOffset = 6;
+  static constexpr size_t kMetadataQueuePosOffset = 6;
+
```

Comment:
Do we need all those constants in the header file?

Reply:
No, I was just trying to be consistent with the other constants above (e.g., kHeaderPosOpcode is defined in the header but could be defined in the source). Let me know if you prefer otherwise.